### PR TITLE
[Backport 2.x] Cleanup interfaces around ExtensionManager (#7374)

### PR DESCRIPTION
**Pull request by peternied**
### Description
ExtensionsManager has a whole lot of functionality that shouldn't be exposed to external classes, marked much of this package protected for a cleaner interface.  Then updated the Noop version to override all public functions to remove extraneous feature flag checks.

Note; I didn't go so far to remove all getters/setters that were unused but that could be a good follow up in the future.

Signed-off-by: Peter Nied <petern@amazon.com>
(cherry picked from commit 3687d0106230186afec58dd5e7b0a58c80f6f7d1)

### Related Issues
- Related #7374

### Check List
- [ ] ~New functionality includes testing.~
  - [ ] ~All tests pass~
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [X] Commits are signed per the DCO using --signoff
- [ ] ~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
FYI @cwperks Once this is merged and built it will unblock https://github.com/opensearch-project/security/pull/2743

**Review by owaiskazi19 — APPROVED** [review]

**Comment by peternied** [comment]
Same test failed 4 times, looks unrelated, to this change, but could be a break on the 2.x line

> #### org.opensearch.search.aggregations.bucket.terms.SignificantTextAggregatorTests.testSupportedFieldTypes
> [ Error Details](javascript:hideFailureSummary('id29091-error'))
> java.lang.AssertionError: Aggregator [significant_text] supports field type [unsigned_long] but executing against the field threw an exception: [Value [0] is out of range for an unsigned long]

**Pull request merged**
